File: LATIHAN2/Latihan2_Java/Hardware.java
public class Hardware extends Product {

	private String brand;
	private String model;

	// empty constuctors
	Hardware() {

	}

	// constructors
	Hardware(String brand, String model) {
		this.brand = brand;
		this.model = model;
	}

	// getter and setter
	public void setBrand(String brand) {
		this.brand = brand;
	}

	public String getBrand() {
		return brand;
	}

	public void setModel(String model) {
		this.model = model;
	}

	public String getModel() {
		return model;
	}

}

File: LATIHAN2/Latihan2_PHP/Hardware.php
<?php

include "Product.php";

// define class hardware inherit from product
class Hardware extends Product
{
	// class attribute
	private $model;
	private $brand;

	// constructors
	public function Hardware($model = "", $brand = "")
	{
		$this->model = $model;
		$this->brand = $brand;
	}

	// getter and setter
	public function setModel($model)
	{
		$this->model = $model;
	}

	public function getModel()
	{
		return $this->model;
	}

	public function setBrand($brand)
	{
		$this->brand = $brand;
	}

	public function getBrand()
	{
		return $this->brand;
	}
}
